pkg/hubble/metrics/metrics.go

```diff
@@ -25,11 +26,18 @@ import (
 	_ "github.com/cilium/cilium/pkg/hubble/metrics/policy"            // invoke init
 	_ "github.com/cilium/cilium/pkg/hubble/metrics/port-distribution" // invoke init
 	_ "github.com/cilium/cilium/pkg/hubble/metrics/tcp"               // invoke init
+	slim_corev1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1"
+)
+
+const (
+	podRemovalBufferSize     = 1024
+	podRemovalGracefulPeriod = time.Minute
 )
 
 var (
 	enabledMetrics *api.Handlers
 	registry       = prometheus.NewPedanticRegistry()
+	podRemovalChan = make(chan *slim_corev1.Pod, podRemovalBufferSize)
```

Comment:
I think we should avoid using global variables at the package level like this. Consts are other insignificant variables are fine at the package level. Would you mind refactoring this to be be part of some struct that we instantiate, etc.?

Reply:
Refactored a little bit, but there is still a place for improvements (eg. setting delay in config options)